spec/controllers/track_controller_spec.rb

```diff
@@ -386,4 +386,50 @@
       expect(tt.track_query).to eq("requested_from:" + geraldine.url_name + " variety:sent")
     end
   end
+
+  describe 'POST delete_all_type' do
+
+    let(:track_thing) { FactoryGirl.create(:search_track) }
+
+    context 'when the user passed in the params is not logged in' do
+
+      it 'redirects to the signin page' do
+        post :delete_all_type, :user => track_thing.tracking_user.id,
+                               :track_type => 'search_query'
+        expect(response).to redirect_to(:controller => 'user',
+                                        :action => 'signin',
+                                        :token => get_last_post_redirect.token)
```

Comment:
`expect(response).to redirect_to(signin_path(:token => get_last_post_redirect.token))` would be nicer here.